src/main/java/fm/last/moji/impl/GetOutputStreamCommand.java

```diff
@@ -62,7 +62,7 @@ public void executeWithTracker(Tracker tracker) throws IOException {
     for (Destination destination : destinations) {
       log.debug("Creating output stream to: {}", destination);
       try {
-        stream = new FileUploadOutputStream(trackerFactory, httpFactory, key, domain, destinations.get(0), writeLock);
+        stream = new FileUploadOutputStream(trackerFactory, httpFactory, key, domain, destination, writeLock);
```

Comment:
Good spot, this looks like a bug that has been in there for ever. I guess in the past this code wasn't in a for loop so just pulled out the first destination but then when it was changed the get(0) was never removed. I wish we could test this somehow, I had a quick look but couldn't come up with something that wouldn't require quite a bit of refactoring. I'm OK with this going in as it is so feel free to merge it (or I can if we'd prefer to have the person raising the PR and the person doing the merge being different).